target/linux/ath79/dts/ar9342_ubnt_lap-120.dts

```diff
@@ -31,8 +31,8 @@
 
 	gmac-config {
 		device = <&gmac>;
-		rxd-delay = <3>;
-		rxdv-delay = <3>;
+		rxd-delay = <2>;
+		rxdv-delay = <2>;
 	};
```

Comment:
You are changing the delays introduced by the MAC - the patch changed the delays introduced by the PHY.

In the end it doesn't matter. Have you verified the link works on every link speed without dropping packets?

Reply:
Yes, good point.

Before changing the delay values, none of the speeds worked (10/100/1000). With this change, I see now it only works for 1000 Mbps. Where to go next? pll-data?

I've also seen that setting rxd and rxdv to <1> also works but, again, only for 1000 Mbps.

Reply:
Did it work previously for all link speeds? If so - you can try out `rgmii-id`, `rgmii-rxid`, `rgmii-txid` on the MACs (not the PHYs!) `phy-mode` property and leave the GMAC delays untouched. One of the should reflect the previous state.

Reply:
Yes, it did work at the three speeds, 10, 100 and 1000 Mbps, with no packet loss.

The stock firmware states RGMII
```
[   36.535000] ATH_MAC_TIMER: enet unit:0 is up (RGMii 10Mbps half duplex)
[   36.535000] ATH_MAC_TIMER: done cfg2 0x7114 ifctl 0x0 miictrl  
[   36.535000] ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
```

I'm trying to find the right combination. `rgmii-id` works for the three speeds, but packet losses appear on 10 and 100 Mbps. `rgmii-rxid` and `rgmii-txid` work for 10 and 100 Mbps, but not for 1 Gbps.

Should I better test the delays or the pll-data?

Reply:
> I'm trying to find the right combination. `rgmii-id` works for the three speeds, but packet losses appear on 10 and 100 Mbps. `rgmii-rxid` and `rgmii-txid` work for 10 and 100 Mbps, but not for 1 Gbps.

Great, rgmii-id works correctly at all speeds; my USB network adapter was causing the retransmissions.